### PR TITLE
FeignClientFactory can be configured with timeout Request.Options

### PR DESCRIPTION
**Pull request by uschi2000**

### CODE REVIEW

**Comment by uschi2000** [comment]
@markelliot @ellisjoe

File: http-clients/src/main/java/com/palantir/remoting/http/FeignClientFactory.java [hunk 1]
```diff
@@ -49,13 +51,15 @@ private FeignClientFactory(
             Decoder decoder,
             ErrorDecoder errorDecoder,
             Function<Optional<SSLSocketFactory>, Client> clientSupplier,
-            BackoffStrategy backoffStrategy) {
+            BackoffStrategy backoffStrategy,
+            Request.Options options) {
```
Comment (markelliot) [inline http-clients/src/main/java/com/palantir/remoting/http/FeignClientFactory.java:55]: should we support a default constructor which does not require providing this parameter to allow compatibility?

Comment (markelliot) [inline http-clients/src/main/java/com/palantir/remoting/http/FeignClientFactory.java:55]: nvm, i get it now

Comment (ellisjoe) [inline http-clients/src/main/java/com/palantir/remoting/http/FeignClientFactory.java:55]: The first .of method is still the same and this constructor is private so we should be ok. The only method we may want is a .of with backoffStrategy and no Request.Options but I'm not sure that's necessary.

Comment (markelliot) [inline http-clients/src/main/java/com/palantir/remoting/http/FeignClientFactory.java:55]: Yeah, I cut the release with this as is, we can add the other method later, as necessary.

**Pull request merged**